annotation_widgets/event_validation/logic.py

```diff
@@ -134,41 +150,50 @@ def load_image(self):
 
         if orig_image is not None:
             self.orig_image = orig_image
-            self.canvas = orig_image
+            self.update_canvas()
 
     def set_video_cap(self):
         video_path = os.path.join(self.pm.videos_path, self.video_names[self.item_id])
         assert video_path.endswith("mp4")
 
-        self.cap = cv2.VideoCapture(video_path)
+        self.frames.clear()
         self.current_frame_number = 0
-        self.number_of_frames = self.cap.get(cv2.CAP_PROP_FRAME_COUNT)
+
+        self.cap = cv2.VideoCapture(video_path)
+        self.number_of_frames = int(self.cap.get(cv2.CAP_PROP_FRAME_COUNT))
```

Comment:
```
def set_video_cap(self, frames_limit: int = 1000):
        video_path = os.path.join(self.pm.videos_path, self.video_names[self.item_id])
        assert video_path.endswith("mp4")

        self.frames.clear()
        self.current_frame_number = 0

        self.cap = cv2.VideoCapture(video_path)

        # Getting frame number via opencv is not a reliable option. It can return 0 or extremely big value, or just incorrect value. It is better to count them direclty or use ffprobe or ffmpeg
        if not self.cap.isOpened():
            raise MessageBoxException(f"Error opening video file {video_path}")

        counter = 0
        while True:
            ret, frame = self.cap.read()
            if ret:
                self.frames.append(frame)
                counter += 1
            else:
                break
            if counter > frames_limit:
                break
        self.number_of_frames = counter
        self.cap.release()
```

Reply:
frames_limit is also worth adding to prevent freezing on unusually large videos that may be encountered

Reply:
Didn't know about that specific. Refactored